### PR TITLE
fix: tree-stop tests fail GitLab INIT_TEST_DATA setup

### PR DESCRIPTION
**Pull request by d-mo**
## Summary
- GitLab unit jobs set `INIT_TEST_DATA=true`. `TestClient` lifespan then seeds via `next(get_db_session()).query(...)`.
- Tree-stop/child-wait tests patched `get_session_factory` to return a context manager only, so that seed saw `_GeneratorContextManager` and every client test ERRORed with `Database setup failed`.
- The bound factory now still shares the test transaction, but also looks like a Session (`query` / no-op `close`), which is what `create_test_data` needs.

## Test plan
- [ ] `pytest backend/tests/test_bound_session.py backend/tests/services/test_flow_tree_stop.py backend/tests/services/test_flow_child_wait.py` with `INIT_TEST_DATA=true` does not ERROR at setup.
- [ ] GitLab `test:unit:preloop-other` after this SHA is in the EE `preloop` submodule: 0 errors in `test_flow_tree_stop.py`.

> [!NOTE]
> **Low Risk**
> Test-harness-only change with direct regression tests; no production code paths are modified.
>
> **Overview**
> Fixes GitLab `INIT_TEST_DATA=true` setup errors in the tree-stop/child-wait tests by replacing a contextmanager-only session-factory stub with a shared `BoundSession` stand-in that still shares the test transaction but exposes Session methods to the lifespan seed (`next(get_db_session()).query` in `create_test_data`). Adds `backend/tests/bound_session.py` plus regression tests. Also widens the log-persistence backpressure test's exception catch to `_RETRYABLE_DB_ERRORS` and raises its wait budgets so self-hosted overflow shards stop timing out.
>
> <sup>Written by [Preloop PR Reviewer](https://preloop.ai) for commit 5f6de3e7. Updates automatically on new commits.</sup>

### CODE REVIEW

File: backend/tests/bound_session.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    """A sessionmaker() result that shares the test transaction.
+
+    ``with factory() as db`` yields the real session and does not close it
+    on exit. ``factory()`` itself still has Session methods so
+    ``get_db_session`` / ``create_test_data`` can ``query`` and ``close``
+    without rolling back the test transaction.
+    """
+
+    def __init__(self, session: Session) -> None:
+        object.__setattr__(self, "_session", session)
```
[the remark this answers is not in the document]
Comment (d-mo) [reply backend/tests/bound_session.py]: Fixed in 5f6de3e. `BoundSession.__init__` now assigns `self._session = session` so mypy sees a `Session`, not `__getattr__` → `object`.
